tests/test-stmt_nested_for.mb

```diff
@@ -0,0 +1,30 @@
+float function determinant(matrix a){
```

Comment:
I feel like you could write a simple nested for loop instead of this complex code to separate the concerns.